### PR TITLE
[test] Revert to using `fireEvent` instead of `userEvent`

### PR DESCRIPTION
**Pull request by LukasTy**
Revert changes introduced in https://github.com/mui/mui-x/pull/14142 in an effort to avoid leaking configuration and flakily failing tests.

There were no flaky `test_unit` failures on this branch. 👌

### CODE REVIEW

**Comment by cherniavskii** [comment]
Why revert these changes though?

**Comment by LukasTy** [comment]
> Why revert these changes though?

@cherniavskii because the amount of flaky Picker test failures has been too intense lately.
It seems that they started after introducing more async tests to Pickers.

I have tried moving most of the Pickers tests to async in https://github.com/mui/mui-x/pull/14583 and it caused way more flaky failures. 🙈 

It seems that we can not reliably mix async and sync tests while we are on mocha because it has trouble isolating test contexts. 🤷

**Comment by LukasTy** [comment]
However, for some reason, I haven't seen `test:unit` fail flakily in the past 10 master merges.

How has it been in everyone's branches? 🤔

**Comment by LukasTy** [comment]
I've converted the PR to a draft until we see a new flaky `test:unit` pipeline failure.

**Comment by LukasTy** [comment]
Here we go again... 🙈 
https://app.circleci.com/pipelines/github/mui/mui-x/68873/workflows/1691361d-2eee-491a-b2e5-a84c73131ae5/jobs/392537
So, the flakiness is not gone, it's just varied. 😆

**Comment by cherniavskii** [comment]
@LukasTy Do you think we can have the `material-ui-v6` workflow passing in this PR?

File: scripts/useMaterialUIv6.mjs [hunk 1]
```diff
@@ -2,7 +2,14 @@ import childProcess from 'child_process';
...
 const pnpmUpdate = childProcess.spawnSync(
   'pnpm',
-  ['update', '-r', '@mui/material@6.x', '@mui/system@6.x', '@mui/icons-material@6.x'],
+  [
+    'update',
+    '-r',
+    '@mui/material@6.x',
+    '@mui/system@6.x',
+    '@mui/icons-material@6.x',
+    '@mui/utils@6.x',
```
Comment (LukasTy) [inline scripts/useMaterialUIv6.mjs:11]: I have added the `@mui/utils` bump to make it more correct. 🤔 
I feel like after https://github.com/mui/mui-x/pull/14867 it makes sense. 🤷

**Comment by LukasTy** [comment]
The PR is ready for review.
Hopefully, it stabilizes the `test_unit` CI step. 🤞

**Review by cherniavskii — APPROVED** [review]
Looks good! 💙

**Pull request merged**